### ISSUE TITLE
NodeJS connection errors and cannot reinstall in Brackets 1.4

### ISSUE THREAD

**Issue by bfrohs** [zaggino/brackets-git#1109]
After upgrading to Brackets 1.4, I began receiving a NodeJS connection error:

> Failed to connect to NodeJS. If you've just updated the extension then close all instances of Brackets and try starting again. If restart doesn't help try removing other extensions that you've installed as one of them is probably causing this problem.

So, I disabled all extensions other than brackets-git, but still receive the error.

I tried to reinstall the extension, but attempting to remove brackets-git results in:

> Unable to remove one or more extensions: zaggino.brackets-git. Brackets will still reload.

However, it is possible to *disable* the extension, which gets rid of the NodeJS error.

---

Brackets Release 1.4 experimental build 1.4.0-16380 (release 273804eb3)

brackets-git release 0.14.27

**Comment by bfrohs** [comment]
I've removed brackets-git manually and attempted to install via the Extension Manager, but receive the following error:

> **Installation failed.**
> Unknown internal error.

The error in the console is:

> Unknown installation error extensionManager domain is undefined - /extensibility/Package.js:410

**Comment by zaggino** [comment]
@bfrohs that seems like an issue of Brackets itself, not the extension, are you able to install other extensions? cc @nethip

**Comment by nethip** [comment]
@bfrohs Unforutnately I am not able to repro this bug. I am able to successfully install/uninstall `brackets-git` extension with 1.4. Could you rename the preferences folder and give it a try?

**Comment by bfrohs** [comment]
> Could you rename the preferences folder and give it a try?

This solved the issue. *sets Brackets back up to how it was* Thanks so much!

**Comment by bfrohs** [comment]
> This solved the issue. sets Brackets back up to how it was Thanks so much!

On further use, this only solved the inability to reinstall brackets-git.

I was still receiving the "Failed to connect to NodeJS" error. However, https://github.com/zaggino/brackets-git/issues/410#issuecomment-101851134 pointed me in the right direction. I commented out additional 127.0.0.1 entries in my hosts file, saved, and reopened Brackets. Everything worked fine. I then went back to my hosts file and uncommented all lines that I just commented, saved again, and reopened Brackets... and there was no issue. So, I'm not really sure *why* this fixed it, but it did.
